pkg/transport/process/utils/uds_server.go

```diff
@@ -0,0 +1,74 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+package utils
+
+import (
+	"io"
+	"log"
+	"net"
+	"sync"
+)
+
+// HandlerFunc defines the interface of a function that should be served by a UDS server
+type HandlerFunc func(io.Reader, io.WriteCloser)
+
+// UDSServer implements a Unix Domain Socket server
+type UDSServer struct {
+	listener net.Listener
+	quit     chan interface{}
+	wg       sync.WaitGroup
+	handler  HandlerFunc
+}
+
+// NewUDSServer returns a new UDS server.
+// The parameters define the server address and the handler func it serves
+func NewUDSServer(addr string, handler HandlerFunc) (*UDSServer, error) {
+	l, err := net.Listen("unix", addr)
+	if err != nil {
+		return nil, err
+	}
+	s := &UDSServer{
+		quit:     make(chan interface{}),
+		listener: l,
+		handler:  handler,
+	}
+	return s, nil
+}
+
+// Start starts the server goroutine
+func (s *UDSServer) Start() {
+	s.wg.Add(1)
+	go s.serve()
+}
+
+func (s *UDSServer) serve() {
+	defer s.wg.Done()
+
+	for {
+		conn, err := s.listener.Accept()
```

Comment:
what happens if s.quit contains a value to indicate stopping. Does listener.Accept() block until a connection arrives and therefore only stops afterwards? Then the stop would not work as expected

Reply:
yes, the Accept() call is blocking. but when s.listener.Close() is called inside the Stop() method from a different thread, the Accept() call will unblock and return with an error.

Reply:
perfect, than it works this way